storage/internal/benchmarks/utils.go

```diff
@@ -0,0 +1,176 @@
+	if min > max {
+		log.Fatalf("min cannot be larger than max; min: %d max: %d", min, max)
+	}
+	return rand.Intn(max-min+1) + min
+}
+
+func randomBucketName(prefix string) string {
+	var sb strings.Builder
+	// The total length of the bucket name must be <= 63 characters
+	maxLen := 63
+	date := time.Now().Format("06-01-02-1504")
+
+	sb.WriteString(prefix)
+	sb.WriteRune('-')
+	sb.WriteString(date)
+	sb.WriteRune('_')
+
+	maxRandomChars := maxLen - sb.Len()
+	sb.WriteString(randomString(maxRandomChars, lowercaseLettersAndNumbers))
+
+	return sb.String()
+}
+
+func randomObjectName() string {
+	// GCS accepts object name up to 1024 characters, but 128 seems long enough to
+	// avoid collisions.
+	maxLen := 128
+
+	return randomString(maxLen, lowercaseLettersAndNumbers+uppercaseLetters)
+}
+
+// random string of maxLen
+func randomString(maxLen int, allowedChars string) string {
+	var sb strings.Builder
+	sb.Grow(maxLen)
+
+	for i := 0; i < maxLen; i++ {
+		sb.WriteByte(allowedChars[rand.Intn(len(allowedChars))])
+	}
+	return sb.String()
+}
+
+// createBenchmarkBucket creates a bucket and returns a function to delete it
+func createBenchmarkBucket(bucketName string, opts *benchmarkOptions) func() {
+	ctx := context.Background()
+
+	// Create a bucket for the tests. We do not need to benchmark this.
+	c, err := storage.NewClient(ctx, option.WithCredentialsFile(credentialsFile))
+	if err != nil {
+		log.Fatalf("NewClient: %v", err)
+	}
+	err = c.Bucket(bucketName).Create(ctx, projectID, &storage.BucketAttrs{
+		Location:     opts.region,
+		StorageClass: "STANDARD",
+	})
+	if err != nil {
+		log.Fatalf("bucket.Create: %v", err)
+	}
+
+	return func() {
+		if err := c.Bucket(bucketName).Delete(context.Background()); err != nil {
+			log.Fatalf("bucket delete: %v", err)
+		}
+	}
+}
+
+func initializeClient(ctx context.Context, api benchmarkAPI, writeBufferSize, readBufferSize int, connPoolSize int) (*storage.Client, benchmarkAPI, benchmarkAPI, error) {
+	var readAPI, writeAPI benchmarkAPI
+	var client *storage.Client
+	var err error
+
+	if api == mixed {
+		if randomBool() {
+			api = xml
+		} else {
+			api = grpc
+		}
+	}
+
+	dialer := &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+
+	// These are the default parameters with write and read buffer sizes modified
+	base := &http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		DialContext:           dialer.DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		WriteBufferSize:       int(writeBufferSize),
+		ReadBufferSize:        int(readBufferSize),
+	}
+	trans, err := htransport.NewTransport(ctx, base, option.WithScopes("https://www.googleapis.com/auth/devstorage.full_control"),
+		option.WithUserAgent("custom-user-agent"), option.WithCredentialsFile(credentialsFile))
```

Comment:
Scope is required to be able to upload/download to gcs. Usually we set this internally but since we are using a custom http client here we must set it manually. 
CredentialsFile is required to pass in custom credentials (which is what I've been doing on my compute instance). We could remove that and require credentials to be set in a different way but I like the simplicity of this.
I removed user agent option as it is not necessary.